fire/core.py

```diff
@@ -127,7 +132,7 @@ def Fire(component=None, command=None, name=None):
     result = component_trace.GetResult()
     print(
         helputils.HelpString(result, component_trace, component_trace.verbose))
-    raise FireExit(1, component_trace)
+    raise FireExit(2, trace)
```

Comment:
good call - fixed in f7c6a46 - I had started working on this branch and then rebased it over your code - my bad on missing it.  I updated the test to check the type of trace too